operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependentResource.java

```diff
@@ -154,7 +154,7 @@ public R update(R actual, R target, P primary, Context<P> context) {
           .forceConflicts().serverSideApply();
     } else {
       var updatedActual = updaterMatcher.updateResource(actual, target, context);
-      updatedResource = prepare(updatedActual, primary, "Updating").replace();
+      updatedResource = prepare(updatedActual, primary, "Updating").update();
```

Comment:
This may need to be an explicitly locked replace, or a patch.  One subtle difference between update and replace is that replace does some modifications to the resource (Services, Jobs, and OpenShift RoleBindings) based upon the present state - see HasMetadataOperation.modifyItemForReplaceOrPatch.  The intention is to remove that once replace is gone.

Reply:
This does not makes up to me, we such changes are done for the resources:

https://github.com/fabric8io/kubernetes-client/blob/0c7d5150702387c1aeca66facb98508d590934f2/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/batch/v1/JobOperationsImpl.java#L163-L175

Shoudn't be this the responsibility of the user to fill those values? 

I don't see why should be this patch or replace instead of update because of this.

Reply:
Yes, it's not clear why these resources have a special treatment to me either…

Reply:
The root issue is that PUT has side effects.  Service is the poster child for this - if you attempt a PUT and the clusterIP is not populated, then it will be allocated, which will then conflict with the existing one.  If you use an empty string it will complain that the field is immutable - people have complained about this for years https://github.com/kubernetes/kubernetes/issues/91459  So I guess that in the past they wanted to smooth this behavior out in the fabric8 client.

In the last couple of years when users complain of new situations like this that don't work with createOrReplace we have been telling to manually do something like the proposed createOr, or more recently to use serverSideApply.

Reply:
Yeah, things can get messy fast when you get into discussion of HTTP verbs semantics :)
I do agree with one of the commenters that PUT should be idempotent so regardless of what controllers do, if they accepted one resource as valid at one point in time, they should accept that same resource again if re-PUT (and possibly return the existing one), which doesn't appear to be the case here…

Reply:
> The root issue is that PUT has side effects. Service is the poster child for this - if you attempt a PUT and the clusterIP is not populated, then it will be allocated, which will then conflict with the existing one. If you use an empty string it will complain that the field is immutable - people have complained about this for years https://github.com/kubernetes/kubernetes/issues/91459 So I guess that in the past they wanted to smooth this behavior out in the fabric8 client.
In the last couple of years when users complain of new situations like this that don't work with createOrReplace we have been telling them to manually do something like the proposed createOr, or more recently to use serverSideApply.


Yeah, I think this workaround for example can nice used:
`First load the existing service that contains the current clusterIP. Set the old clusterIp to the updated V1Service.`

For these special cases would rather prepare some default implementations in dependent resources, rather than solving it on client level here. So would anyways stick with the `update`.